### PR TITLE
BI-2121 - Workflows Select

### PR DESCRIPTION
**Pull request by nickpalladino**
# Description
**Story:** [BI-2121](https://breedinginsight.atlassian.net/browse/BI-2121?atlOrigin=eyJpIjoiY2EyMzNmNmUwNTc5NDk2Yjk4OTU4OWVlYjg4OWVjMmYiLCJwIjoiaiJ9)

- Retrieve workflows from backend and display in select box, call appropriate endpoint with selected workflow id

# Dependencies
- bi-api feature/BI-2122

# Testing
- Verify that the experiment import works the same as it has prior to this card regardless of what workflow is selected (just spot check, if it runs)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have tested my code and ensured it meets the acceptance criteria of the story
- [ ] I have create/modified unit tests to cover this change
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to documentation
- [ ] I have run TAF: _\<link to TAF run>_

[BI-2121]: https://breedinginsight.atlassian.net/browse/BI-2121?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/breeding-insight/dao/ImportDAO.ts [hunk 1]
```diff
@@ -47,6 +47,14 @@ export class ImportDAO {
     return new BiResponse(data);
   }
 
+  static async getWorkflowsForMapping(mappingId: string | undefined) : Promise<BiResponse> {
+    const { data } =  await api.call({
+      url: `${process.env.VUE_APP_BI_API_V1_PATH}/import/mappings/${mappingId}/workflows`,
+      method: 'get'
+    }) as Response;
+    return new BiResponse(data);
+  }
```
Comment (mlm483) [inline src/breeding-insight/dao/ImportDAO.ts:56]: Can one mapping have many workflows associated with it?

File: src/breeding-insight/dao/ImportDAO.ts [hunk 2]
```diff
@@ -99,8 +107,13 @@ export class ImportDAO {
     return new BiResponse(data);
   }
 
-  static async updateUploadData(programId: string, mappingId: string, uploadId: string, userInput: any, commit: boolean) {
+  static async updateUploadData(programId: string, mappingId: string, uploadId: string, workflowId: string | undefined, userInput: any, commit: boolean) {
     let url = `${process.env.VUE_APP_BI_API_V1_PATH}/programs/${programId}/import/mappings/${mappingId}/data/${uploadId}`;
+
+    if (workflowId !== undefined) {
+      url = `${process.env.VUE_APP_BI_API_V1_PATH}/programs/${programId}/import/mappings/${mappingId}/workflows/${workflowId}/data/${uploadId}`;
+    }
```
Comment (mlm483) [inline src/breeding-insight/dao/ImportDAO.ts:115]: I think it makes sense that only one workflow can be interacted with at once. Unless there is a plan to chain multiple workflows into pipelines? Probably wouldn't make sense to make that transparent to the user.

**Review by mlm483 — APPROVED** [review]

File: src/breeding-insight/dao/ImportDAO.ts (hunk 1 above)
Comment (nickpalladino) [reply src/breeding-insight/dao/ImportDAO.ts:56]: Ya, the experiment import mapping has three workflows associated with it

File: src/breeding-insight/dao/ImportDAO.ts (hunk 2 above)
Comment (nickpalladino) [reply src/breeding-insight/dao/ImportDAO.ts:115]: Ya, I think a single workflow at this level makes sense. Maybe for the updating of datasets there could internally be different workflows for sub obs unit vs exp unit that are conditionally kicked off by the general append / update workflow.

**Review by dmeidlin — APPROVED** [review]

**Pull request merged**